discovery/kubernetes/kubernetes.go

```diff
@@ -258,103 +267,138 @@ const resyncPeriod = 10 * time.Minute
 // Run implements the discoverer interface.
 func (d *Discovery) Run(ctx context.Context, ch chan<- []*targetgroup.Group) {
 	d.Lock()
-	namespaces := d.getNamespaces()
+	var namespaces []string
+	if d.memberRollNamespace != "" && d.memberRollController != nil {
+		namespaces = []string{d.memberRollNamespace}
+	} else {
+		namespaces = d.getNamespaces()
+	}
 
 	switch d.role {
 	case RoleEndpoint:
-		for _, namespace := range namespaces {
-			e := d.client.CoreV1().Endpoints(namespace)
-			elw := &cache.ListWatch{
-				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-					return e.List(options)
+		elw := func(namespace string) cache.ListerWatcher {
+			return &cache.ListWatch{
+				ListFunc: func(opts metav1.ListOptions) (runtime.Object, error) {
+					return d.client.CoreV1().Endpoints(namespace).List(opts)
 				},
-				WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-					return e.Watch(options)
+				WatchFunc: func(opts metav1.ListOptions) (watch.Interface, error) {
+					return d.client.CoreV1().Endpoints(namespace).Watch(opts)
 				},
 			}
-			s := d.client.CoreV1().Services(namespace)
-			slw := &cache.ListWatch{
-				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-					return s.List(options)
+		}
+		emnlw := listwatch.MultiNamespaceListerWatcher(namespaces, elw)
```

Comment:
can we use a better name than this? All of these three/four letter variable names make the code harder to parse. Same for each of the three/four letter acronyms being added here.

Reply:
The acronyms follow the original upstream istio naming, we can revisit this at a later time